### PR TITLE
More Python docstrings and minor usability improvements for Jupyter

### PR DESCRIPTION
**Pull request by amueller**
Changing `__str__` to `__repr__` and some other slight polish

### CODE REVIEW

**Comment by amueller** [comment]
hm not sure about the change in the dimension repr, should check with Adam.

**Comment by bpkroth** [comment]
> 
> 
> hm not sure about the change in the dimension repr, should check with Adam.

@byte-sculptor

File: source/Mlos.Python/mlos/Grpc/BayesianOptimizerFactory.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def create_remote_optimizer(self, optimization_problem: OptimizationProblem, opt
 
 
         Returns
-        =======
+        -------
```
Comment (bpkroth) [inline source/Mlos.Python/mlos/Grpc/BayesianOptimizerFactory.py:45]: Are the fixups resulting from the WARNING output from the sphinx script?

Comment (amueller) [reply source/Mlos.Python/mlos/Grpc/BayesianOptimizerFactory.py:45]: No, they are are results of me having had more coffee. We should probably fix the warnings. There are probably >100, we can error once we fixed them.

**Review by bpkroth — APPROVED** [review]
LGTM

**Comment by byte-sculptor** [comment]
Looks good to me!

**Pull request merged**